### ISSUE TITLE
Fix Proguard/R8 rules

### ISSUE THREAD

**Issue by Thomas-Vos** [getsentry/sentry-java#1976]
### Integration

sentry-android

### Build System

Gradle

### AGP Version

all

### Proguard

Enabled

### Version

6.0.0-alpha.4

### Steps to Reproduce

The Proguard/R8 rules keep far too many classes. See for example the following file: 

https://github.com/getsentry/sentry-java/blob/6.x.x/sentry-android-core/proguard-rules.pro

Lots of those rules seem to be needed for GSON, but GSON was removed in 6.x.x. So the rules can be reduced.

Specifically the following:

https://github.com/getsentry/sentry-java/blob/4414ba0f40bd97fa8a1e8879d8dd6b387b4af86b/sentry-android-core/proguard-rules.pro#L15-L18

This should either be changed to keep specific classes, or just removed completely as GSON is no longer used.

Same for the other rules, some of them seem like they were added for GSON but never removed.

### Expected Result

Remove a lot of the rules

### Actual Result

Too many classes are kept

**Comment by marandaneto** [comment]
@Thomas-Vos indeed, it's in our backlog to find out what really needs to be excluded from minification or not, e.g., every public API class, interfaces, classes used in reflection, etc.

**Comment by Thomas-Vos** [comment]
I am using Sentry in a JVM desktop app with R8 obfuscation. With the alpha version mentioned I did not need to add any additional rules for Sentry.

**Comment by marandaneto** [comment]
> I am using Sentry in a JVM desktop app with R8 obfuscation. With the alpha version mentioned I did not need to add any additional rules for Sentry.

Yep, nothing needs to be added, it's about removing the not needed rules anymore.

**Comment by Thomas-Vos** [comment]
Yes, what I meant was that I did not need to add any rules at all. The rules in the Android module were not included in my project as I only depend on the Java dependency.

**Comment by philipphofmann** [comment]
Was fixed with #2031
